### ISSUE TITLE
In a mutual block, yet undefined things should be assumed to be metavariables, not postulates

### ISSUE THREAD

**Issue by GoogleCodeExporter** [agda/agda#1113]
```agda
-- I have a record with two fields, the second field depending on the first one
-- via a construction [B] (that I don’t want to inline inside [C] because it’s a
-- complicated construction in my actual use case)

record R : Set₁ where
  constructor c
  field
    A : Set

  B : Set
  B = A

  field
    C : B → Set

-- Now I want to define an element of [R] by giving [A] and [C] (and without
-- having to inline the definition of [B] anywhere)
-- The following seems to be the simplest way to do it (but I may have missed something)

r : R
r = r'  where

  r' : R
  A' : Set
  C' : R.B r' → Set

  r' = c A' {!C'!}
  A' = {!!}
  C' = {!!}

{-
Now if I give the [C'] in [r'], it’s accepted, but if I load the file again, 
Agda
complains with the following error (highlighting the [C']):

/tmp/Test.agda:27,13-15
A' !=< (R.A r') of type Set
when checking that the expression C' has type A' → Set
-}
```

Original issue reported on code.google.com by `guillaum...@gmail.com` on 30 Apr 2014 at 1:20

**Comment by Saizan** [comment]
The bug is that give accepts the definition, as in #118.

Considering later definitions as metas would probably lead to very(insanely) dependent types, so it's not clear if we want to. See #1556 .

**Comment by UlfNorell** [comment]
Closing. See @Saizan's comment about insanely dependent types.

**Comment by jespercockx** [comment]
It seems that the fix of https://github.com/agda/agda/issues/5600 in https://github.com/agda/agda/commit/982e18cdcbb2355d571bc05a8a98811fe6b48525 by @Saizan has changed the behaviour of Agda on this issue. Now Agda accepts the OP in this issue with the hole filled in even after a reload:
```agda
record R : Set₁ where
  constructor c
  field
    A : Set
    C : A → Set

r : R
A : Set
C : R.A r → Set

r = c A C
A = {!!}
C = {!!}
```
Given the previous discussion here, this is not behaving as intended. Hence I propose to revert the fix of #5600 in order to re-fix this issue (and also its +5000 successor #6113). @UlfNorell @andreasabel what do you think?

**Comment by andreasabel** [comment]
Mmh, I do not directly see a harm in accepting this example.  However, #6008 is annoying. One could say that #5672 (fix of #5600) caused regressions and thus is a candidate for revert.
